app/core/src/main/java/uk/ac/ebi/fg/annotare2/core/magetab/MageTabGenerator.java

```diff
@@ -518,17 +561,16 @@ public Integer apply(Protocol protocol) {
         for (Protocol protocol : orderedProtocols) {
             // protocol node name must be unique
             String nodeName = prev.getClass().getSimpleName() + ":" + prev.getNodeName() + ":" + protocol.getId() + (protocol.isAssigned() ? "" : "F");
-            ProtocolApplicationNode protocolNode = getNode(ProtocolApplicationNode.class, nodeName);
-            if (null == protocolNode) {
-                if (protocol.isAssigned()) {
-                    protocolNode = createNode(ProtocolApplicationNode.class, nodeName);
-                    protocolNode.protocol = protocol.getName();
-                    if (protocol.hasPerformer()) {
-                        PerformerAttribute attr = new PerformerAttribute();
-                        attr.setAttributeValue(protocol.getPerformer());
-                        protocolNode.performer = attr;
-                    }
+            ProtocolApplicationNode protocolNode = null;
+            if (protocol.isAssigned()) {
+                protocolNode = createNode(ProtocolApplicationNode.class, nodeName);
```

Comment:
Can `createNode` return `null`?

Reply:
Yes, `createNode` will never return `null`. I have simplified this block by removing unnecessary null checks.
But here we need additional variable `protocolNode` to hold current node and link it to 'prev' which is source node. And then `prev' has been changed to current node.